arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -124,26 +125,25 @@ void DatabaseManagerThread::run() {
 
   while (true) {
     try {
-      TRI_vocbase_t* database = nullptr;
+      // if we find a database to delete, it will be automatically deleted at
+      // the end of this scope
+      std::unique_ptr<TRI_vocbase_t> database;
       {
         std::lock_guard lock{databaseFeature._databasesMutex};
         auto& dropped = databaseFeature._droppedDatabases;
         // check if we have to drop some database
         for (auto it = dropped.begin(), end = dropped.end(); it != end; ++it) {
           TRI_ASSERT(*it != nullptr);
           if ((*it)->isDangling()) {
-            database = *it;
+            database.reset(*it);
             dropped.erase(it);
```

Comment:
I'm not sure, is erase noexcept (on practice)?
Anyway it can be easy fixed:
auto* ptr = *it;
erase(it)
database.reset(ptr);